src/Tribe/Attendee_Repository.php

```diff
@@ -663,4 +688,414 @@ protected function limit_list( $key, $list ) {
 			$key => $list[ $key ],
 		];
 	}
+
+	/**
+	 * {@inheritDoc}
+	 *
+	 * @return WP_Post|false The new post object or false if unsuccessful.
+	 */
+	public function create() {
+		/*
+		 * Only create if we are using a specific attendee context. The post type used is
+		 * entirely dependent on the provider-specific implementation for attendees.
+		 */
+		if ( ! $this->key_name ) {
+			return false;
+		}
+
+		/*
+		 * Only create if we have a ticket set.
+		 */
+		if ( ! isset( $this->updates['ticket_id'] ) ) {
+			return false;
+		}
+
+		return parent::create();
+	}
+
+	/**
+	 * Create an attendee object from ticket and attendee data.
+	 *
+	 * @since TBD
+	 *
+	 * @param Tribe__Tickets__Ticket_Object $ticket        The ticket object.
+	 * @param array                         $attendee_data List of additional attendee data.
+	 *
+	 * @return WP_Post|false The new post object or false if unsuccessful.
+	 *
+	 * @throws Tribe__Repository__Usage_Error If the argument types are not set as expected.
+	 */
+	public function create_attendee_for_ticket( $ticket, $attendee_data ) {
```

Comment:
Maybe we can make it more flexible if we allow the `$ticket` parameter to contemplate the case where they send the `$ticket_id`? And add a first if to see if it's an int, to look for the ticket object?

Reply:
Done, originally this wasn't going to be possible but I expanded each extended class to provide an attendee provider and reuse it elsewhere so it can look that up now